content/en/llm_observability/instrumentation/sdk.md

```diff
@@ -1793,6 +1795,105 @@ def rag_workflow(user_question):
 {{% /tab %}}
 {{< /tabs >}}
 
+## Prompt tracking
+
+Attach structured prompt metadata to the LLM span so you can reproduce results, audit changes, and compare prompt performance across versions. When using templates, LLM Observability also provides [version tracking](#version-tracking) based on template content changes.
+
+### Inline annotation (Python)
+
+Use `LLMObs.annotation_context(prompt=...)` to attach prompt metadata immediately before the LLM call. Use this for one-off or ad‑hoc prompts inside a function. The metadata is added to the current active span (for example, a span from a decorator or an auto-instrumented LLM call). For more details on span annotation, see [Annotating a span](#annotating-a-span).
+
+`prompt`
+: required - dictionary
+<br />A typed dictionary that follows the Prompt schema below.
+
+{{% collapse-content title="Prompt structure" level="h4" expanded=false id="prompt-structure" %}}
+
+Supported keys:
+
+- `id` (string): Logical identifier for this prompt. Should be unique per `ml_app`. Defaults to `{ml_app}-unnamed_prompt`
+- `version` (string): Version tag for the prompt (for example, "1.0.0"). See [version tracking](#version-tracking) for more details.
+- `variables` (object: Dict[str, str]): Variables used to render the template.
+- `template` (string): Single string template form.
+- `chat_template` (list of objects or Messages): Multi-message template form. Provide a list of `{ "role": "<role>", "template": "<template>" }` objects, or a list of `Message` objects.
+- `tags` (object: Dict[str, str]): Tags to attach to the prompt run.
+- `rag_context_variables` (list of strings): Variable keys that contain ground-truth/context content. Used for [hallucination detection](/llm_observability/evaluations/managed_evaluations/?tab=openai#hallucination).
+- `rag_query_variables` (list of strings): Variable keys that contain the user query. Used for [hallucination detection](/llm_observability/evaluations/managed_evaluations/?tab=openai#hallucination).
+
+{{% /collapse-content %}}
+
+Notes:
+- Annotating a prompt is only available on LLM spans.
+- Place the annotation immediately before the provider call so it applies to the correct LLM span.
```

Comment:
these two bullet points are crucial and why we should lead with an example here bc this is too easy to get wrong IMO